docs/config/@config.md

```diff
@@ -0,0 +1,222 @@
+
+@Config Use
+-----------
+
+This annotation is used to denote a class is the container for some configuration options. 
+
+There are 4 properties:
+
+|     Property|Type     | Default Value | Comment 
+|           -:|:-       |     :-:       |:-
+|        modid|`String` |     N/A       | The mod id that this configuration is associated with. 
+|         name|`String` |     `""`      | A user friendly name for the config file, the default will be modid. 
+|         type|`Type`   |`Type.INSTANCE`| The type this is, right now the only value is `Type.INSTANCE`. This is intended to be expanded upon later for more Forge controlled configs. 
+|     category|`String` |  `"general"`  | Root element category. If this is an empty string then the root category is disabled. 
+
+!!! important
+  If you disable the root category, this will cause issues unless you create [Sub Categories][subcategories].
+
+@Comment Use
+------------
+
+Adding a `@Comment` annotation to a field will add a comment to the config in its file. 
+
+It has 1 property:
+
+| Property|Type       | Default Value |Comment 
+|       -:|:-         |       :-:     |:-
+|    value|`String[]` |       N/A     |A `String[]` will form a multi-line comment where 1 element = 1 line.
+!!! note
+  The value in this annotation can be passed as a `String` because java is cool and turns it into a `String[]`.
+
+### Example
+```java
+@Comment({
+  "You can add comments using this",
+  "and if you supply an array it will be multi-line"
+})
+public static boolean doTheThing = true;
+```
+This would produce the following config:
+```
+# You can add comments using this
+# and if you supply an array it will me multiline
+B:doTheThing=true   
+``` 
+This will produce a `boolean` config value with name `doTheThing`, the defualt value `true` and the comment provided in the config file.
+
+@Name Use
+---------
+
+You should use the `@Name` annotation to give a config a user friendly name in the config file.
+
+This has 1 property:
+| Property | Type    | Default Value |
+|        -:|:-       |      :-:      |
+|     value|`String` |      N/A      | 
+
+### Example
+```java
+@Name("FE/T for the thing")
+public static int thingFE = 50; 
+```
+This will produce the following config:
+```
+I:"FE/T for the thing"=50
+```
+
+@RangeInt Use
+-------------
+
+You should use `@RangeInt` to limit an `int` or `Integer` config value. 
+
+It has a couple of properties:
+
+| Property|Type   |    Default Value    |
+|       -:|:-     |        :-:          |
+|      min|`int`  | `Integer.MIN_VALUE` |
+|      max|`int`  | `Integer.MAX_VALUE` |
+
+### Example
+```java
+@RangeInt(min = 0) 
+public static int thingFECapped = 50;
+```
+This will produce the following config:
+
+```
+# Min: 0
+# Max: 2147483647
+I:thingFECapped=50
+```
+
+@RangeDouble Use
+----------------
+
+You should use `@RangeDouble` to limit a `double` or `Double` config value. 
+
+It has a couple of properties:
```

Comment:
Explicitly state 2 instead of "a couple". You do this throughout the above types.